FreeTubeApp/FreeTube
# Improve video playlist search UX

# Improve video playlist search UX




## Pull Request Type


## Related issue
None

## Description
Clicking a button to enable/disable search has felt pretty arbitrary and cumbersome in my experience. This PR:
1. Removes the playlist video search button & makes the search bar appear alongside the other `playlist-info` controls by default, where `playlistInVideoSearchMode` is now true when the search bar is non-empty.
2. Focuses the playlist search bar on Ctrl+F. This is an experimental change testing how people like the convenience of Ctrl+F for a route with a prominent search bar. I, personally, have been pressing it in my personal usage on this page thinking it would focus this bar, to minor frustration.

## Screenshots 

![Screenshot_20240411_091333](https://github.com/FreeTubeApp/FreeTube/assets/84899178/f4a27796-3a13-43ca-932e-5f5cd5125bd9)


## Testing 
- (REG) Ensure search filter is populated when searching for a video from the User Playlists page
- Ensure search filter is active/inactive based on if trimmed search input is empty
- Try using the icons with search filter active (I thought this was a reason why we had this implemented like this, but it turns out all of the playlist options work just the same)

## Desktop

- **OS:** OpenSUSE
- **OS Version:** TW
